### ISSUE TITLE
Unhandled errors on Client connection failure?

### ISSUE THREAD

**Issue by ethanent** [PrismarineJS/node-minecraft-protocol#538]
My code looks a bit like this:

```js
let client = mc.createClient({
	host, // host (defined elsewhere)
	port, // port (defined elsewhere)
	'username': 'testacc',
	'keepAlive': false,
	'version': false
})

client.on('error', (err) => {
	console.error('Connection error: ' + err)
	client.end('Onticord > An error occurred while routing to a server.')
})

client.socket.on('error', (err) => {
	console.error('Connection error: ' + err)
	client.end('Onticord > An error occurred while routing to a server.')
})
```

So I'm trying to handle connection issues, but when the connection can't be established, the whole Node process crashes.

```
Connection error: Error: connect ECONNREFUSED 127.0.0.1:25566
Connection error: Error: connect ECONNREFUSED 127.0.0.1:25566
/Users/ethan/Stuff/Projects/onticord/onticord/node_modules/minecraft-protocol/src/client/autoVersion.js:14
    if (err) throw err; // hmm
             ^

Error: connect ECONNREFUSED 127.0.0.1:25566
    at Object._errnoException (util.js:1031:13)
    at _exceptionWithHostPort (util.js:1052:20)
    at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1195:14)
```

**Comment by oscartbeaumont** [comment]
It is running the console.error statement and closing because there is no more code to run. You need to put some code in where you have the console.error statements so that it does not close.

**Comment by rom1504** [comment]
The bug is that throw.
We should be doing client.error instead

**Comment by rom1504** [comment]
I mean client.emit('error',...)

**Comment by ethanent** [comment]
👍  That's what I figured. Thought it was something I was doing and that I was going crazy. 😆

**Comment by ethanent** [comment]
Patched with merged PR #539!
